src/wallet/wallet.h

```diff
@@ -177,28 +178,33 @@ class CMerkleTx : public CTransaction
      */
     int nIndex;
 
-    CMerkleTx()
+    template<typename... Args>
+    CMerkleTx(Args&&... args)
     {
+        SetTx(std::forward<Args>(args)...);
         Init();
     }
 
-    CMerkleTx(const CTransaction& txIn) : CTransaction(txIn)
-    {
-        Init();
-    }
+    operator const CTransaction&() const { return *tx; }
 
     void Init()
     {
         hashBlock = uint256();
         nIndex = -1;
     }
 
+    template<typename... Args>
+    void SetTx(Args&&... args)
```

Comment:
I've changed CMerkleTx and CWalletTx to just take a `CTransactionRef` argument now as you suggested, effectively moving the `MakeTransactionRef` call to the constructor callers. By doing so, I discovered a few places where an std::move could be introduced to avoid a copy, so that makes the change worth it already.